sympy/geometry/parabola.py

```diff
@@ -0,0 +1,359 @@
+        >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
+        >>> p1.p_parameter
+        -4
+
+        """
+        if (self.axis_of_symmetry.slope == 0):
+            x = -(self.directrix.coefficients[2])
+            if (x < self.focus.args[0]):
+                p = self.focal_length
+            else:
+                p = -self.focal_length
+        else:
+            y = -(self.directrix.coefficients[2])
+            if (y > self.focus.args[1]):
+                p = -self.focal_length
+            else:
+                p = self.focal_length
+
+        return p
+
+    @property
+    def vertex(self):
+        """The vertex of the parabola.
+
+        Returns
+        =======
+
+        vertex : Point
+
+        See Also
+        ========
+
+        sympy.geometry.point.Point
+
+        Examples
+        ========
+
+        >>> from sympy import Parabola, Point, Line
+        >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
+        >>> p1.vertex
+        Point2D(0, 4)
+
+        """
+        focus = self.focus
+        if (self.axis_of_symmetry.slope == 0):
+            vertex = Point(focus.args[0] - self.p_parameter, focus.args[1])
+        else:
+            vertex = Point(focus.args[0], focus.args[1] - self.p_parameter)
+
+        return vertex
+
+    @property
+    def eccentricity(self):
+        """The eccentricity of the parabola.
+
+        Returns
+        =======
+
+        eccentricity : number
+
+        A parabola may also be characterized as a conic section with an
+        eccentricity of 1. As a consequence of this, all parabolas are
+        similar, meaning that while they can be different sizes,
+        they are all the same shape.
+
+        See Also
+        ========
+
+        https://en.wikipedia.org/wiki/Parabola
+
+
+        Examples
+        ========
+
+        >>> from sympy import Parabola, Point, Line
+        >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
+        >>> p1.eccentricity
+        1
+
+        Notes
+        -----
+        The eccentricity for every Parabola is 1 by definition.
+
+        """
+        return S(1)
+
+    def equation(self, x=None, y=None):
+        """The equation of the parabola.
+
+        Parameters
+        ==========
+        x : Symbol, optional
+            Label for the x-axis. Default value is Symbol("x").
+        y : Symbol, optional
+            Label for the y-axis. Default value is Symbol("y").
+
+        Returns
+        =======
+        equation : sympy expression
+
+        Examples
+        ========
+
+        >>> from sympy import Parabola, Point, Line, Symbol
+        >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
+        >>> p1.equation()
+        -x**2 - 16*y + 64
+        >>> p1.equation(Symbol("f"))
+        -f**2 - 16*y + 64
+        >>> p1.equation(y = Symbol("z"))
+        -x**2 - 16*z + 64
+
+        """
+        if x is None:
+            x = Symbol("x")
```

Comment:
> Why sympify('x')? That should give the same thing.

I thought sympify handles a more general set of inputs than Symbol. But I suppose you only want symbol-like objects here. The `_symbol()` function was here earlier (and in the other geometry classes) and that makes the symbols have the real assumption, which complicates the issues in #11043 a bit  more.

> The equation method here should indeed match the other stuff in the geometry module.

The design of this is not that great in my opinion. Especially wrt to the issue in #11043.

> Were the other equation methods added before or after sympy 1.0?

Before I believe.